src/main/java/trycb/service/TenantUser.java

```diff
@@ -21,80 +23,85 @@
         this.jwtService = jwtService;
     }
 
     static final String USERS_COLLECTION_NAME = "users";
-    static final String FLIGHTS_COLLECTION_NAME = "flights";
+    static final String BOOKINGS_COLLECTION_NAME = "bookings";
 
     /**
-     * Try to log the given user in.
+     * Try to log the given tenant user in.
      */
-    public Map<String, Object> login(final Bucket bucket, final String username, final String password) {
+    public Result<Map<String, Object>> login(final Bucket bucket, final String tenant, final String username,
+            final String password) {
+        Scope scope = bucket.scope(tenant);
+        Collection collection = scope.collection(USERS_COLLECTION_NAME);
+        String queryType = String.format("KV get - scoped to %s.users: for password field in document %s", scope.name(),
+                username);
+
         GetResult doc;
         try {
-            doc = bucket.defaultCollection().get(username);
+            doc = collection.get(username);
         } catch (DocumentNotFoundException ex) {
-            throw new AuthenticationCredentialsNotFoundException("Bad Username or Password: " + username);
+            throw new AuthenticationCredentialsNotFoundException("Bad Username or Password");
         }
         JsonObject res = doc.contentAsObject();
-        if(BCrypt.checkpw(password, res.getString("password"))) {
-            return JsonObject.create()
-                .put("token", jwtService.buildToken(username))
-                .toMap();
+        if (BCrypt.checkpw(password, res.getString("password"))) {
+            Map<String, Object> data = JsonObject.create().put("token", jwtService.buildToken(username)).toMap();
+            return Result.of(data, queryType);
         } else {
             throw new AuthenticationCredentialsNotFoundException("Bad Username or Password");
         }
     }
 
     /**
-     * Create a user.
+     * Create a tenant user.
      */
-    public Result<Map<String, Object>> createLogin(final Bucket bucket, final String username, final String password,
-            DurabilityLevel expiry) {
+    public Result<Map<String, Object>> createLogin(final Bucket bucket, final String tenant, final String username,
+            final String password, DurabilityLevel expiry) {
         String passHash = BCrypt.hashpw(password, BCrypt.gensalt());
-        JsonObject doc = JsonObject.create()
-            .put("type", "user")
-            .put("name", username)
-            .put("password", passHash);
+        JsonObject doc = JsonObject.create().put("type", "user").put("name", username).put("password", passHash);
         InsertOptions options = insertOptions();
         if (expiry.ordinal() > 0) {
             options.durability(expiry);
         }
-        String narration = "User account created in document " + username + " in bucket " + bucket.name()
-                + (expiry.ordinal() > 0 ? ", with expiry of " + expiry.ordinal() + "s" : "");
```

Comment:
this narration has gone?

Reply:
I updated this to match the narration we use for Python and Nodejs to keep the output we send to the FE the same:

See line [81](https://github.com/couchbaselabs/try-cb-java/pull/16/files/18b76fd03c4f5e8f09d0c4a03b1c952a92546105..114920fda6d26e41ebea1af3324c32c2e027f5cd#diff-982bc6359b182c2a5bb7995bf0fa11e9b6f31a4ead60046b44a2730e55fa4d2fR81)